src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -364,7 +358,7 @@ int BATT_SMBUS::dataflash_write(uint16_t &address, void *data, const unsigned le
 {
 	uint8_t code = BATT_SMBUS_MANUFACTURER_BLOCK_ACCESS;
 
-	uint8_t tx_buf[DATA_BUFFER_SIZE + 2] = {};
+	uint8_t tx_buf[MAC_DATA_BUFFER_SIZE + 2] = {};
 
 	tx_buf[0] = ((uint8_t *)&address)[0];
 	tx_buf[1] = ((uint8_t *)&address)[1];
```

Comment:
The line below (`memcpy(&tx_buf[2], data, length);`) can also produce an overflow if `length > MAC_DATA_BUFFER_SIZE`. Can you check for that?